### DISCUSSION TITLE
What is the rationale behind the default `exclude` pattern for config files?

### DISCUSSION THREAD

**Discussion by aentwist** [vitest-dev/vitest#6918] [Feedback]
The default `exclude` pattern is

```
'**/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build}.config.*'
```

Naturally, by design this cannot cover everything. It has been said that `**/*.config.*` is too broad. OK, interesting opinion.

It has also been said that adding new config files to this list will be rejected in favor of using `include` instead, for a whitelist instead of blacklist style pattern. Well that does make sense. But then why mislead users by excluding *any* config files?

Why can't I open a PR for Tailwind to be added to this list, even though I use literally *2* of the tools currently on it, and Tailwind has 5M more weekly downloads than Cypress (literally double)?

If the answer is it's basically deprecated, I understand.

It is very misleading that when running coverage (which copies this config exclude pattern from the main option) all of my vite config files are magically excluded, while my Tailwind file isn't. Disappointingly, it sent me on the wrong search path which ended up in GitHub issues.

**Comment by sheremet-va** [comment]
There was no logic when it was initially added. I think it just what made sense at the time and no one oppose it or noticed it.

We don't approve new values added to it because we will remove all these configs in the next major.

**Comment by AriPerkkio** [comment]
Seeing issues likes this makes me think maybe we should try similar approach as Jest has: https://jestjs.io/docs/configuration#collectcoveragefrom-array

It requires user to define `include` pattern - there is no `exclude` option at all. By default it shows only files that were executed (like Vitest with `coverage.all: false`). Users have to define uncovered files in `collectCoverageFrom` so that they come up in report.
